pmlb/pmlb.py

```diff
@@ -137,3 +137,89 @@ def get_updated_datasets(local_cache_dir='datasets'):
     return {'changed_datasets': changed_datasets,
             'changed_metadatas': changed_metadatas}
 
+from sklearn.neighbors import NearestNeighbors
+from sklearn.preprocessing import StandardScaler
+from .update_dataset_files import generate_summarystats
+from pathlib import Path
+import numpy as np
+
+
+def fetch_nearest_dataset_names(X,y=None, **kwargs):
+    """
+    X: numpy array
+        an n_samples x n_features array of independent variables
+    y: numpy array or None (default: None)
+        a n_samples array of dependent variables
+    """
+    df = pd.DataFrame({**{'x_'+str(i):x for i,x in enumerate(X.transpose)}
+                       **{'target':y}})
+    return fetch_nearest_dataset_names(df, **kwargs)
+
+def fetch_nearest_dataset_names(df, n=1, 
+        dimensions=['#instances','#features'],
+        task=None):
+    """Returns names of most similar datasets to df, in order of similarity. 
+
+    Parameters
+    ----------
+    df: pandas Dataframe 
+        a dataframe of n_samples x n_features+1 with a target column labeled
+        'target'
+    n: int (default: 1)
+        the number of dataset names to return
+    dimensions: list of str or str (default: ['NumberOfInstances',
+    'NumberOfFeatures'])
+        a list of dataset characteristics to include in similarity calculation.
+        Dimensions must correspond to columns of datasets/all_summary_stats.csv.
+        If 'all', uses all numeric columns.
+    task: str or None (default: None)
+        specify classification or regression for summary stat generation. If 
+        None, we use classification unless the target column has more than 5 
+        unique values.
+
+    Returns
+    -------
+    dataset_names: an n-element list of dataset names in order of most similar 
+        to least similar.
+    """
+    # get summary stats for dataset
+    if task==None:
+        task = 'classification' if df['target'].nunique()<5 else 'regression'
```

Comment:
Can we just use the problem_type column from `pmlb_stats` below? If we want to redefine `task`, I think we can edit it [here](https://github.com/EpistasisLab/penn-ml-benchmarks/blob/4e84eeb75adf6f80d319c007d65562e37ac97902/pmlb/update_dataset_files.py#L112).

Reply:
Please disregard.